bench/app.py

```diff
@@ -371,22 +405,75 @@ def get_app(
 		app.install(verbose=verbose, skip_assets=skip_assets)
 
 
+def install_resolved_deps(
+	bench,
+	resolution,
+	bench_path=".",
+	skip_assets=False,
+	verbose=False,
+):
+	from bench.utils.app import check_existing_dir
+
+	if "frappe" in resolution:
+		# Terminal dependency
+		del resolution["frappe"]
+
+	for repo_name, app in reversed(resolution.items()):
+		existing_dir, path_to_app = check_existing_dir(bench_path, repo_name)
+		if existing_dir:
+			is_compatible = False
+
+			try:
+				installed_branch = bench.apps.states[repo_name]["resolution"]["branch"].strip()
+			except:
+				installed_branch = (
+					subprocess.
```

Comment:
Check scope for using the `gitpython` library here

Reply:
Git gives a lot more flexibility with ease so I think it would be better to use git directly here?